### PR TITLE
Update the internal global TextMapPropagator

### PR DESCRIPTION
**Pull request by MrAlias**
Include feedback from @seh in a post-merge review of #1258

### CODE REVIEW

**Comment by MrAlias** [comment]
@seh looks like you're not a member of @open-telemetry so I can't request a review from you in the automation, but would appreciate your feedback here.

**Review by seh — COMMENTED** [review]
This looks good.

One alternative to consider: Introduce a function like this:
```go
func (p *textMapPropagator) effectiveDelegate() otel.TextMapCarrier {
  if d := p.currentDelegate(); d != nil {
    return d
  }
  return p.noop
}
```
Then, in the three cases here, you would call it, for example,
```go
  return p.effectiveDelegate().Inject(ctx, carrier)
```

**Comment by seh** [comment]
> looks like you're not a member of @open-telemetry

What does it take to become a member? Where do I send my check?

**Comment by MrAlias** [comment]
> What does it take to become a member? Where do I send my check?

:laughing: 

Instructions are [here](https://github.com/open-telemetry/community/blob/master/community-membership.md#member). Please consider me as one of your sponsors when you submit your application.

File: api/global/internal/propagator.go [hunk 1]
```diff
@@ -54,34 +54,29 @@ func (p *textMapPropagator) SetDelegate(delegate otel.TextMapPropagator) {
 	p.mtx.Unlock()
 }
 
-// currentDelegate returns the current delegate of p in a concurrently-safe
-// manner.
-func (p *textMapPropagator) currentDelegate() otel.TextMapPropagator {
+// effectiveDelegate returns the current delegate of p if one is set,
+// otherwise the default noop TextMapPropagator is returned. This method is
```
Comment (seh) [inline api/global/internal/propagator.go:58]:
```suggestion
// otherwise the default noop TextMapPropagator is returned. This method
```

**Review by seh — APPROVED** [review]
My vote doesn't count, but this looks good to me, modulo the one extra word in the documentation comment.

**Comment by Aneurysm9** [comment]
> > What does it take to become a member? Where do I send my check?
> 
> 😆
> 
> Instructions are [here](https://github.com/open-telemetry/community/blob/master/community-membership.md#member). Please consider me as one of your sponsors when you submit your application.

Count me as a second sponsor!

**Review by Aneurysm9 — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Comment by seh** [comment]
>> Please consider me as one of your sponsors when you submit your application.
> 
> Count me as a second sponsor!

It's been a while, but please see open-telemetry/community#695.

**Pull request merged**